### PR TITLE
Adding information on file versioning to FAQ

### PR DESCRIPTION
**Pull request by mkaramlou**
### What change does this PR introduce and why?
Adds to the FAQ about the ability to support file versioning. 
KOL-3953

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ N/A] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: docs/faq/index.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ If you don't see your question here, please reach out to us on Slack or at
     We're constantly adding new data types and annotation types — if you don't see what you're looking for, reach out
     to us and we'll happily extend our system to support your use case.
 
+??? faq "Does Kolena support file versioning?"
+
+    If you are using Amazon S3 or Google Cloud Storage, Kolena supports file versioning for your test samples. Simply
+    enable bucket versioning on your [S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Versioning.html) or [GCS](https://cloud.google.com/storage/docs/object-versioning)
```
Comment (gordonhart) [inline docs/faq/index.md]: Suggested copy update:

> Simply enable versioning on your S3 or GCS bucket and make sure ...

File: docs/faq/index.md [hunk 2]
```diff
@@ -28,6 +28,12 @@ If you don't see your question here, please reach out to us on Slack or at
...
+    and make sure to pass the version id as part of the locator. See the examples in [`kolena.workflow.TestSample`][kolena.workflow.TestSample]
```
Comment (gordonhart) [inline docs/faq/index.md]: Add a period at the end of the sentence

File: docs/faq/index.md [hunk 3]
```diff
@@ -28,6 +28,12 @@ If you don't see your question here, please reach out to us on Slack or at
     We're constantly adding new data types and annotation types — if you don't see what you're looking for, reach out
     to us and we'll happily extend our system to support your use case.
 
+??? faq "Does Kolena support file versioning?"
```
Comment (gordonhart) [inline docs/faq/index.md]: I'd move this to the "Using Kolena" section below — this topmost "About Kolena" section is intended to answer the high-level questions about what we do.

**Comment by mkaramlou** [comment]
@gordonhart addressed the comments and updated the PR. Let me know if there is anything else.

**Review by gordonhart — APPROVED** [review]

**Pull request merged**